### PR TITLE
fix(emptyState): set the correct size modifier using the  variant prop

### LINKED ISSUE #2680: PF4 Possible bug on empty state size class binding
Looking at the demo page of the Empty State component ( https://www.patternfly.org/v4/documentation/react/components/emptystate/  ) it looks like the css class for the component size is not being generated correctly, on the demo page, the small size one is being generated as **'styles.modifiers.sm'** and the large one as **'styles.modifiers.lg'**, if i'm not wrong they should be generated as **'pf-m-sm'** and **'pf-m-lg'** ( as in the HTML demo page ).

### PR DESCRIPTION
**Pull request by boaz0**
**What**:

closes #2680 

//cc @jessiehuff can you look at this, am I missing something? :pray: 
//cc @carlosga can you look at this?

### CODE REVIEW

**Comment by carlosga** [comment]
Looks good to me, thanks very much, my only question would be if it ¿shouldn't be using the getModifier function? ( as in other components ), something like:

export enum EmptyStateVariant {
  large = 'lg',
  small = 'sm',
  full = ''
}

...

export const EmptyState: React.FunctionComponent<EmptyStateProps> = ({
  children,
  className = '',
  variant = EmptyStateVariant.large,
  ...props
}: EmptyStateProps) => {
  return (
    <div className={css(styles.emptyState, getModifier(styles, variant), className)} {...props}>
      {children}
    </div>
  );
};

________________________________
De: boaz0 <notifications@github.com>
Enviado: martes, 13 de agosto de 2019 0:25
Para: patternfly/patternfly-react <patternfly-react@noreply.github.com>
Cc: Carlos Guzmán Álvarez <carlos.guzman.1@outlook.es>; Mention <mention@noreply.github.com>
Asunto: [patternfly/patternfly-react] fix(emptyState): set the correct size modifier using the variant prop (#2692)


What:

closes #2680<https://github.com/patternfly/patternfly-react/issues/2680>

//cc @jessiehuff<https://github.com/jessiehuff> can you look at this, am I missing something? 🙏
//cc @carlosga<https://github.com/carlosga> can you look at this?

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/patternfly/patternfly-react/pull/2692

Commit Summary

  *   fix(emptyState): set the correct size modifier using the variant prop

File Changes

  *   M packages/patternfly-4/react-core/src/components/EmptyState/EmptyState.tsx<https://github.com/patternfly/patternfly-react/pull/2692/files#diff-0> (19)
  *   M packages/patternfly-4/react-core/src/components/EmptyState/__snapshots__/EmptyState.test.tsx.snap<https://github.com/patternfly/patternfly-react/pull/2692/files#diff-1> (4)

Patch Links:

  *   https://github.com/patternfly/patternfly-react/pull/2692.patch
  *   https://github.com/patternfly/patternfly-react/pull/2692.diff

**Comment by boaz0** [comment]
@carlosga - that's a good idea. let me try and do it.

File: packages/patternfly-4/react-core/src/components/EmptyState/EmptyState.tsx [hunk 1]
```diff
@@ -5,21 +5,22 @@ import styles from '@patternfly/react-styles/css/components/EmptyState/empty-sta
...
+  full: ''
+};
+
 export interface EmptyStateProps extends React.HTMLProps<HTMLDivElement> {
   /** Additional classes added to the EmptyState */
   className?: string;
   /** Content rendered inside the EmptyState */
   children: React.ReactNode;
   /** Modifies EmptyState max-width */
-  variant?: 'small' | 'large' | 'full';
```
Comment (jschuler) [inline packages/patternfly-4/react-core/src/components/EmptyState/EmptyState.tsx]: For docs purposes we decided to type out the variant options

File: packages/patternfly-4/react-core/src/components/EmptyState/EmptyState.tsx [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
-  full = 'full',
-};
-
-export enum maxWidthModifiers {
-  large = 'styles.modifiers.lg',
-  small = 'styles.modifiers.sm'
+  full = 'full'
 }
 
+const maxWidthModifiers: { [variant in keyof typeof EmptyStateVariant]: string } = {
```
Comment (redallen) [inline packages/patternfly-4/react-core/src/components/EmptyState/EmptyState.tsx]: Thanks for the extra type safety.

**Comment by boaz0** [comment]
Thank you! 
//cc @jschuler @redallen @carlosga

**Pull request merged**